azavea/geotrellis-collections-api-research
# Paint Pennsylania NLCD Tiles

## Overview

Adds:

 - A painting class to paint tiles from the ingested Pennsylvania NLCD raster. It picks up colors from the original GeoTIFF's embedded values. It has a number of comments to help follow the code.
 - `make` commands to allow for painting
 - A tile toggle button to the front-end, and the ability to render locally generated tiles in the front-end when toggled on
 - Relevant new information to the README

Updates:

 - Minor readability improvements
 - Upgrade GeoTrellis to 2.0.0-RC1
 - Change default `make` to also paint tiles
 - Conditionally run `download-tif` and `ingest` commands if their output doesn't already exist

Removes:

 - Unused imports from the ingest class

Connects https://github.com/azavea/civic-apps/issues/313

### Notes

Once this PR is merged, I'll make a follow up that updates the README GIF with a new version that also showcases the painted tiles.

Maybe we up the version from `1.0` to something else? `1.1`? `2.0`?

I had to update the source TIF which was missing coordinate information to get the ingest and painting to work correctly (see discussion here: https://azavea.slack.com/archives/C049A6HC9/p1530903555000168). This new source TIF should be uploaded to S3 to replace the current one.

I should probably add the run condition to the `make paint` command as well so it is not extraneously run.

## Testing Instructions

* Checkout this branch
* Get the updated source TIF from me
* Run `make` and ensure it completes successfully
* Run `make server` and ensure you can see the tiles by turning them on in the header
